File: General/increasing_triplet_subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        if(size(nums)<3){
            return false;
        }
        int x = INT_MAX, y = INT_MAX;
        for (int i=0; i<size(nums); i++){
            if(nums[i]<=x){
                x=nums[i];
            }
            else if(nums[i]<=y){
                y=nums[i];
            }
            else{
                return true;
            }
        }
        return false;
    }
};

File: General/increasing_triplet_subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        if(len(nums)<3):
            return False
        
        x=max(nums)+1        
        y=max(nums)+1
        
        for num in nums:
            if(num<=x):
                x=num
            elif(num<=y):
                y=num
            else:
                return True
        
        return False